app/main/posts/views/post-filters.service.js

```diff
@@ -30,6 +31,14 @@ function PostFiltersService(_, FormEndpoint, TagEndpoint, $q) {
         qEnabled: false
     };
 
+    /**
+     * function to deal with the fact that logout and login don't really reset the defaults.
+     */
+    function resetDefaults() {
+        return activate().then(function () {
```

Comment:
just run activate to reset all the tags and such, and then run clearfilters which uses the defaults previously set

Reply:
@crcommons some context on why this is async and why this is not a broadcast